scripts/DeployAgTokenSideChainMultiBridge.s.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "forge-std/Script.sol";
+import "utils/src/CommonUtils.sol";
+import { AgTokenSideChainMultiBridge } from "contracts/agToken/AgTokenSideChainMultiBridge.sol";
+import { LayerZeroBridgeToken } from "contracts/agToken/layerZero/LayerZeroBridgeToken.sol";
+import { ICoreBorrow } from "contracts/interfaces/ICoreBorrow.sol";
+import { Treasury } from "contracts/treasury/Treasury.sol";
+import { ImmutableCreate2Factory } from "contracts/interfaces/external/create2/ImmutableCreate2Factory.sol";
+
+contract DeployAgTokenSideChainMultiBridge is Script, CommonUtils {
+    using stdJson for string;
+
+    string constant JSON_VANITY_PATH = "./scripts/vanity.json";
+
+    function run() external {
+        /** TODO  complete */
+        string memory stableName = vm.envString("STABLE_NAME");
+        address expectedAddress = vm.envAddress("EXPECTED_ADDRESS");
+        uint256 totalLimit = vm.envUint("TOTAL_LIMIT");
+        uint256 hourlyLimit = vm.envUint("HOURLY_LIMIT");
+        uint256 chainTotalHourlyLimit = vm.envUint("CHAIN_TOTAL_HOURLY_LIMIT");
+        bool mock = vm.envOr("MOCK", false);
+        /** END  complete */
+
+        uint256 deployerPrivateKey = vm.deriveKey(vm.envString("MNEMONIC_MAINNET"), "m/44'/60'/0'/0/", 0);
+        address deployer = vm.addr(deployerPrivateKey);
+        string memory json = vm.readFile(JSON_VANITY_PATH);
+        bytes32 salt = bytes32(abi.encodePacked(deployer, abi.encodePacked(uint96(json.readUint("$.init")))));
+        uint256 chainId = vm.envUint("CHAIN_ID");
+
+        address proxyAdmin = _chainToContract(chainId, ContractType.ProxyAdmin);
+        address coreBorrow = _chainToContract(chainId, ContractType.CoreBorrow);
+        ILayerZeroEndpoint lzEndpoint = _lzEndPoint(chainId);
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        AgTokenSideChainMultiBridge agTokenImpl = new AgTokenSideChainMultiBridge();
+        console.log("AgTokenSideChainMultiBridge Implementation deployed at", address(agTokenImpl));
+
+        ImmutableCreate2Factory create2Factory = ImmutableCreate2Factory(IMMUTABLE_CREATE2_FACTORY_ADDRESS);
+        bytes memory initCode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(address(agTokenImpl), proxyAdmin, "")
```

Comment:
need to put a 0 address for the agTokenImpl and the deployer for the proxyAdmin otherwise you won't be able to always keep the same address